packages/api/src/task/scheduler.ts

```diff
@@ -264,6 +277,36 @@ export class TaskScheduler {
     await this.updateTask(task, { status });
   }
 
+  async retryTask(task: WithID<Task>, event?: messages.TaskResult) {
+    let attempts = task.status.attempts + 1 ?? 1;
+    if (attempts <= MAX_RETRY_ATTEMPTS) {
+      const status: Task["status"] = {
+        ...task.status,
+        attempts: attempts,
+      };
+
+      await this.updateTask(task, { status });
+      task.status = status;
+
+      if (attempts > 1) {
+        // No timeout at first retry, 15 seconds * attempt after that
+        let retryDelay = attempts * TASK_RETRY_DELAY_COEFFICIENT;
+        await new Promise((resolve) => setTimeout(resolve, retryDelay));
+      }
+
+      await this.enqueueTask(task);
+      return true;
+    } else {
+      console.log(
+        `task retry process error: err=max retries reached taskId=${task.id}`
+      );
+      if (event) {
```

Comment:
Is there any case this function will be called without an event? 0o

Reply:
:eyes: I figured it may be possible, but in any case I am going to remove this check with the failTask from this function and handle it on top as you suggested